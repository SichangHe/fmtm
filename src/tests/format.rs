use super::*;

fn default_format(input: &str) -> Result<String, std::fmt::Error> {
    format(input, Some(80))
}

fn format_20(input: &str) -> Result<String, std::fmt::Error> {
    format(input, Some(20))
}

fn format_inf(input: &str) -> Result<String, std::fmt::Error> {
    format(input, None)
}

macro_rules! t {
    ($name:ident, $input:expr) => {
        #[test]
        fn $name() {
            init_tracing();
            let input = $input.trim_start();
            let default_formatted = default_format(input).unwrap();
            assert_snapshot!(default_formatted);
            let formatted_20 = format_20(input).unwrap();
            assert_snapshot!(formatted_20);
            let formatted_inf = format_inf(input).unwrap();
            assert_snapshot!(formatted_inf);
        }
    };
}

t!(gpt_inline, include_str!("gpt_inline.md"));

t!(gpt1, include_str!("gpt1.md"));
