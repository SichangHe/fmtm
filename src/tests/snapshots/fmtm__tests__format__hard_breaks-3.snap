---
source: src/tests/format.rs
expression: formatted_inf
---
Markdown documents may contain hard breaks.\
For example, the one above.

\
Best,\
You Know Who
